### ISSUE TITLE
Huion Mini Keydial KD100 

### ISSUE THREAD

**Issue by albfan** [linuxwacom/libwacom#469]
device:
![image](https://user-images.githubusercontent.com/220968/151121623-9c2d7df6-0893-492a-a2a3-e56a8231bc4b.png)

receiver:
![image](https://user-images.githubusercontent.com/220968/151121829-d37eadae-0021-4cd3-93b4-538cca1adab5.png)

**Device name**
Huion Mini Keydial KD100 

**Device model identifier**
This is problematic, as this device uses an usb to connect wirelessly the dongle expose:

```
Bus 001 Device 005: ID 256c:006d
```

if device is connected wired it shows:

```
Bus 001 Device 011: ID 256c:006d HUION Huion Tablet_KD100
```

projects like https://github.com/kurikaesu/userspace-tablet-driver-daemon read firmware to detect which device is connected:

```
std::wstring huion_tablet::getDeviceFirmwareName(libusb_device_handle *handle) {
    auto *buffer = new unsigned char[200];
    memset(buffer, 0, 200);

    // Extract the firmware name
    int descriptorLength = libusb_get_string_descriptor(handle, 0xc9, 0x0409, buffer, 130);
```

```
std::string huion_tablet::getDeviceNameFromFirmware(std::wstring firmwareName) {
    // HUION tablets
    if (firmwareName == L"HUION_T188_180718") {
        return "Huion WH1409 v2";
    } else if (firmwareName == L"HUION_T191_190619") {
        return "Huion H1161";
    } else if (firmwareName == L"HUION_T153_160524") {
        return "Huion WH1409 (2048)";
    } else if (firmwareName == L"HUION_T200_210309") {
        return "Huion KD100 mini Keydial";
```

Is it possible to add this matching style?

**udevadm info output**

It exposes js0, mouse* input* but about event:

```
event17 -> ../../devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.0/0003:256C:006D.0024/input/input98/event17
event18 -> ../../devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.0/0003:256C:006D.0024/input/input99/event18
event19 -> ../../devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.1/0003:256C:006D.0025/input/input101/event19
event20 -> ../../devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.1/0003:256C:006D.0025/input/input102/event20
event21 -> ../../devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.2/0003:256C:006D.0026/input/input103/event21
event22 -> ../../devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.2/0003:256C:006D.0026/input/input104/event22

$ udevadm info /sys/class/input/event17
P: /devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.0/0003:256C:006D.0024/input/input98/event17
N: input/event17
L: 0
S: input/by-id/usb-256c_006d-event-mouse
S: input/by-path/pci-0000:00:14.0-usb-0:1:1.0-event-mouse
E: DEVPATH=/devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.0/0003:256C:006D.0024/input/input98/event17
E: DEVNAME=/dev/input/event17
E: MAJOR=13
E: MINOR=81
E: SUBSYSTEM=input
E: USEC_INITIALIZED=26553909035
E: ID_INPUT=1
E: ID_INPUT_TABLET=1
E: ID_VENDOR=256c
E: ID_VENDOR_ENC=256c
E: ID_VENDOR_ID=256c
E: ID_MODEL=006d
E: ID_MODEL_ENC=006d
E: ID_MODEL_ID=006d
E: ID_REVISION=0100
E: ID_SERIAL=256c_006d
E: ID_TYPE=hid
E: ID_BUS=usb
E: ID_USB_INTERFACES=:030102:030000:
E: ID_USB_INTERFACE_NUM=00
E: ID_USB_DRIVER=usbhid
E: ID_PATH=pci-0000:00:14.0-usb-0:1:1.0
E: ID_PATH_TAG=pci-0000_00_14_0-usb-0_1_1_0
E: LIBINPUT_DEVICE_GROUP=3/256c/6d:usb-0000:00:14.0-1
E: DEVLINKS=/dev/input/by-id/usb-256c_006d-event-mouse /dev/input/by-path/pci-0000:00:14.0-usb-0:1:1.0-event-mouse

$ udevadm info /sys/class/input/event18
P: /devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.0/0003:256C:006D.0024/input/input99/event18
N: input/event18
L: 0
S: input/by-path/pci-0000:00:14.0-usb-0:1:1.0-event-mouse
S: input/by-id/usb-256c_006d-event-mouse
E: DEVPATH=/devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.0/0003:256C:006D.0024/input/input99/event18
E: DEVNAME=/dev/input/event18
E: MAJOR=13
E: MINOR=82
E: SUBSYSTEM=input
E: USEC_INITIALIZED=26553912549
E: ID_INPUT=1
E: ID_INPUT_TABLET=1
E: ID_INPUT_TABLET_PAD=1
E: ID_VENDOR=256c
E: ID_VENDOR_ENC=256c
E: ID_VENDOR_ID=256c
E: ID_MODEL=006d
E: ID_MODEL_ENC=006d
E: ID_MODEL_ID=006d
E: ID_REVISION=0100
E: ID_SERIAL=256c_006d
E: ID_TYPE=hid
E: ID_BUS=usb
E: ID_USB_INTERFACES=:030102:030000:
E: ID_USB_INTERFACE_NUM=00
E: ID_USB_DRIVER=usbhid
E: ID_PATH=pci-0000:00:14.0-usb-0:1:1.0
E: ID_PATH_TAG=pci-0000_00_14_0-usb-0_1_1_0
E: LIBINPUT_DEVICE_GROUP=3/256c/6d:usb-0000:00:14.0-1
E: DEVLINKS=/dev/input/by-path/pci-0000:00:14.0-usb-0:1:1.0-event-mouse /dev/input/by-id/usb-256c_006d-event-mouse

$ udevadm info /sys/class/input/event19
P: /devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.1/0003:256C:006D.0025/input/input101/event19
N: input/event19
L: 0
S: input/by-id/usb-256c_006d-if01-event-kbd
S: input/by-path/pci-0000:00:14.0-usb-0:1:1.1-event-kbd
E: DEVPATH=/devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.1/0003:256C:006D.0025/input/input101/event19
E: DEVNAME=/dev/input/event19
E: MAJOR=13
E: MINOR=83
E: SUBSYSTEM=input
E: USEC_INITIALIZED=26553871134
E: ID_INPUT=1
E: ID_INPUT_KEY=1
E: ID_INPUT_KEYBOARD=1
E: ID_VENDOR=256c
E: ID_VENDOR_ENC=256c
E: ID_VENDOR_ID=256c
E: ID_MODEL=006d
E: ID_MODEL_ENC=006d
E: ID_MODEL_ID=006d
E: ID_REVISION=0100
E: ID_SERIAL=256c_006d
E: ID_TYPE=hid
E: ID_BUS=usb
E: ID_USB_INTERFACES=:030102:030000:
E: ID_USB_INTERFACE_NUM=01
E: ID_USB_DRIVER=usbhid
E: ID_PATH=pci-0000:00:14.0-usb-0:1:1.1
E: ID_PATH_TAG=pci-0000_00_14_0-usb-0_1_1_1
E: ID_INPUT_TABLET=0
E: LIBINPUT_DEVICE_GROUP=3/256c/6d:usb-0000:00:14.0-1
E: DEVLINKS=/dev/input/by-id/usb-256c_006d-if01-event-kbd /dev/input/by-path/pci-0000:00:14.0-usb-0:1:1.1-event-kbd
E: TAGS=:power-switch:
E: CURRENT_TAGS=:power-switch:

$ udevadm info /sys/class/input/event20
P: /devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.1/0003:256C:006D.0025/input/input102/event20
N: input/event20
L: 0
S: input/by-id/usb-256c_006d-event-if01
S: input/by-path/pci-0000:00:14.0-usb-0:1:1.1-event
E: DEVPATH=/devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.1/0003:256C:006D.0025/input/input102/event20
E: DEVNAME=/dev/input/event20
E: MAJOR=13
E: MINOR=84
E: SUBSYSTEM=input
E: USEC_INITIALIZED=26553876896
E: ID_INPUT=1
E: ID_INPUT_KEY=1
E: ID_VENDOR=256c
E: ID_VENDOR_ENC=256c
E: ID_VENDOR_ID=256c
E: ID_MODEL=006d
E: ID_MODEL_ENC=006d
E: ID_MODEL_ID=006d
E: ID_REVISION=0100
E: ID_SERIAL=256c_006d
E: ID_TYPE=hid
E: ID_BUS=usb
E: ID_USB_INTERFACES=:030102:030000:
E: ID_USB_INTERFACE_NUM=01
E: ID_USB_DRIVER=usbhid
E: ID_PATH=pci-0000:00:14.0-usb-0:1:1.1
E: ID_PATH_TAG=pci-0000_00_14_0-usb-0_1_1_1
E: LIBINPUT_DEVICE_GROUP=3/256c/6d:usb-0000:00:14.0-1
E: DEVLINKS=/dev/input/by-id/usb-256c_006d-event-if01 /dev/input/by-path/pci-0000:00:14.0-usb-0:1:1.1-event
E: TAGS=:power-switch:
E: CURRENT_TAGS=:power-switch:

$ udevadm info /sys/class/input/event21
P: /devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.2/0003:256C:006D.0026/input/input103/event21
N: input/event21
L: 0
S: input/by-path/pci-0000:00:14.0-usb-0:1:1.2-event-mouse
S: input/by-id/usb-256c_006d-if02-event-mouse
E: DEVPATH=/devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.2/0003:256C:006D.0026/input/input103/event21
E: DEVNAME=/dev/input/event21
E: MAJOR=13
E: MINOR=85
E: SUBSYSTEM=input
E: USEC_INITIALIZED=26553915994
E: ID_INPUT=1
E: ID_INPUT_MOUSE=1
E: ID_VENDOR=256c
E: ID_VENDOR_ENC=256c
E: ID_VENDOR_ID=256c
E: ID_MODEL=006d
E: ID_MODEL_ENC=006d
E: ID_MODEL_ID=006d
E: ID_REVISION=0100
E: ID_SERIAL=256c_006d
E: ID_TYPE=hid
E: ID_BUS=usb
E: ID_USB_INTERFACES=:030102:030000:
E: ID_USB_INTERFACE_NUM=02
E: ID_USB_DRIVER=usbhid
E: ID_PATH=pci-0000:00:14.0-usb-0:1:1.2
E: ID_PATH_TAG=pci-0000_00_14_0-usb-0_1_1_2
E: ID_INPUT_TABLET=1
E: LIBINPUT_DEVICE_GROUP=3/256c/6d:usb-0000:00:14.0-1
E: DEVLINKS=/dev/input/by-path/pci-0000:00:14.0-usb-0:1:1.2-event-mouse /dev/input/by-id/usb-256c_006d-if02-event-mouse

$ udevadm info /sys/class/input/event22
P: /devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.2/0003:256C:006D.0026/input/input104/event22
N: input/event22
L: 0
S: input/by-path/pci-0000:00:14.0-usb-0:1:1.2-event
S: input/by-id/usb-256c_006d-event-if02
E: DEVPATH=/devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.2/0003:256C:006D.0026/input/input104/event22
E: DEVNAME=/dev/input/event22
E: MAJOR=13
E: MINOR=86
E: SUBSYSTEM=input
E: USEC_INITIALIZED=26553983213
E: ID_INPUT=1
E: ID_INPUT_TOUCHSCREEN=1
E: ID_INPUT_WIDTH_MM=9
E: ID_INPUT_HEIGHT_MM=9
E: ID_VENDOR=256c
E: ID_VENDOR_ENC=256c
E: ID_VENDOR_ID=256c
E: ID_MODEL=006d
E: ID_MODEL_ENC=006d
E: ID_MODEL_ID=006d
E: ID_REVISION=0100
E: ID_SERIAL=256c_006d
E: ID_TYPE=hid
E: ID_BUS=usb
E: ID_USB_INTERFACES=:030102:030000:
E: ID_USB_INTERFACE_NUM=02
E: ID_USB_DRIVER=usbhid
E: ID_PATH=pci-0000:00:14.0-usb-0:1:1.2
E: ID_PATH_TAG=pci-0000_00_14_0-usb-0_1_1_2
E: ID_INPUT_TABLET=1
E: LIBINPUT_DEVICE_GROUP=3/256c/6d:usb-0000:00:14.0-1
E: DEVLINKS=/dev/input/by-path/pci-0000:00:14.0-usb-0:1:1.2-event /dev/input/by-id/usb-256c_006d-event-if02

```

Connected wired it shows:

```
/sys/class/input/event17 -> ../../devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.0/0003:256C:006D.0027/input/input105/event17
/sys/class/input/event18 -> ../../devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.0/0003:256C:006D.0027/input/input106/event18
/sys/class/input/event19 -> ../../devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.1/0003:256C:006D.0028/input/input108/event19
/sys/class/input/event20 -> ../../devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.1/0003:256C:006D.0028/input/input109/event20

$ udevadm info /sys/class/input/event17
P: /devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.0/0003:256C:006D.0027/input/input105/event17
N: input/event17
L: 0
S: input/by-path/pci-0000:00:14.0-usb-0:1:1.0-event-mouse
S: input/by-id/usb-HUION_Huion_Tablet_KD100-event-mouse
E: DEVPATH=/devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.0/0003:256C:006D.0027/input/input105/event17
E: DEVNAME=/dev/input/event17
E: MAJOR=13
E: MINOR=81
E: SUBSYSTEM=input
E: USEC_INITIALIZED=26885167808
E: ID_INPUT=1
E: ID_INPUT_TABLET=1
E: ID_VENDOR=HUION
E: ID_VENDOR_ENC=HUION
E: ID_VENDOR_ID=256c
E: ID_MODEL=Huion_Tablet_KD100
E: ID_MODEL_ENC=Huion\x20Tablet_KD100
E: ID_MODEL_ID=006d
E: ID_REVISION=0100
E: ID_SERIAL=HUION_Huion_Tablet_KD100
E: ID_TYPE=hid
E: ID_BUS=usb
E: ID_USB_INTERFACES=:030102:
E: ID_USB_INTERFACE_NUM=00
E: ID_USB_DRIVER=usbhid
E: ID_PATH=pci-0000:00:14.0-usb-0:1:1.0
E: ID_PATH_TAG=pci-0000_00_14_0-usb-0_1_1_0
E: LIBINPUT_DEVICE_GROUP=3/256c/6d:usb-0000:00:14.0-1
E: DEVLINKS=/dev/input/by-path/pci-0000:00:14.0-usb-0:1:1.0-event-mouse /dev/input/by-id/usb-HUION_Huion_Tablet_KD100-event-mouse

$ udevadm info /sys/class/input/event18
P: /devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.0/0003:256C:006D.0027/input/input106/event18
N: input/event18
L: 0
S: input/by-id/usb-HUION_Huion_Tablet_KD100-event-mouse
S: input/by-path/pci-0000:00:14.0-usb-0:1:1.0-event-mouse
E: DEVPATH=/devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.0/0003:256C:006D.0027/input/input106/event18
E: DEVNAME=/dev/input/event18
E: MAJOR=13
E: MINOR=82
E: SUBSYSTEM=input
E: USEC_INITIALIZED=26885161664
E: ID_INPUT=1
E: ID_INPUT_TABLET=1
E: ID_INPUT_TABLET_PAD=1
E: ID_VENDOR=HUION
E: ID_VENDOR_ENC=HUION
E: ID_VENDOR_ID=256c
E: ID_MODEL=Huion_Tablet_KD100
E: ID_MODEL_ENC=Huion\x20Tablet_KD100
E: ID_MODEL_ID=006d
E: ID_REVISION=0100
E: ID_SERIAL=HUION_Huion_Tablet_KD100
E: ID_TYPE=hid
E: ID_BUS=usb
E: ID_USB_INTERFACES=:030102:
E: ID_USB_INTERFACE_NUM=00
E: ID_USB_DRIVER=usbhid
E: ID_PATH=pci-0000:00:14.0-usb-0:1:1.0
E: ID_PATH_TAG=pci-0000_00_14_0-usb-0_1_1_0
E: LIBINPUT_DEVICE_GROUP=3/256c/6d:usb-0000:00:14.0-1
E: DEVLINKS=/dev/input/by-id/usb-HUION_Huion_Tablet_KD100-event-mouse /dev/input/by-path/pci-0000:00:14.0-usb-0:1:1.0-event-mouse

$ udevadm info /sys/class/input/event19
P: /devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.1/0003:256C:006D.0028/input/input108/event19
N: input/event19
L: 0
S: input/by-path/pci-0000:00:14.0-usb-0:1:1.1-event-kbd
S: input/by-id/usb-HUION_Huion_Tablet_KD100-if01-event-kbd
E: DEVPATH=/devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.1/0003:256C:006D.0028/input/input108/event19
E: DEVNAME=/dev/input/event19
E: MAJOR=13
E: MINOR=83
E: SUBSYSTEM=input
E: USEC_INITIALIZED=26885123050
E: ID_INPUT=1
E: ID_INPUT_KEY=1
E: ID_INPUT_KEYBOARD=1
E: ID_VENDOR=HUION
E: ID_VENDOR_ENC=HUION
E: ID_VENDOR_ID=256c
E: ID_MODEL=Huion_Tablet_KD100
E: ID_MODEL_ENC=Huion\x20Tablet_KD100
E: ID_MODEL_ID=006d
E: ID_REVISION=0100
E: ID_SERIAL=HUION_Huion_Tablet_KD100
E: ID_TYPE=hid
E: ID_BUS=usb
E: ID_USB_INTERFACES=:030102:
E: ID_USB_INTERFACE_NUM=01
E: ID_USB_DRIVER=usbhid
E: ID_PATH=pci-0000:00:14.0-usb-0:1:1.1
E: ID_PATH_TAG=pci-0000_00_14_0-usb-0_1_1_1
E: ID_INPUT_TABLET=0
E: LIBINPUT_DEVICE_GROUP=3/256c/6d:usb-0000:00:14.0-1
E: DEVLINKS=/dev/input/by-path/pci-0000:00:14.0-usb-0:1:1.1-event-kbd /dev/input/by-id/usb-HUION_Huion_Tablet_KD100-if01-event-kbd
E: TAGS=:power-switch:
E: CURRENT_TAGS=:power-switch:

$ udevadm info /sys/class/input/event20
P: /devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.1/0003:256C:006D.0028/input/input109/event20
N: input/event20
L: 0
S: input/by-id/usb-HUION_Huion_Tablet_KD100-event-if01
S: input/by-path/pci-0000:00:14.0-usb-0:1:1.1-event
E: DEVPATH=/devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.1/0003:256C:006D.0028/input/input109/event20
E: DEVNAME=/dev/input/event20
E: MAJOR=13
E: MINOR=84
E: SUBSYSTEM=input
E: USEC_INITIALIZED=26885115677
E: ID_INPUT=1
E: ID_INPUT_KEY=1
E: ID_VENDOR=HUION
E: ID_VENDOR_ENC=HUION
E: ID_VENDOR_ID=256c
E: ID_MODEL=Huion_Tablet_KD100
E: ID_MODEL_ENC=Huion\x20Tablet_KD100
E: ID_MODEL_ID=006d
E: ID_REVISION=0100
E: ID_SERIAL=HUION_Huion_Tablet_KD100
E: ID_TYPE=hid
E: ID_BUS=usb
E: ID_USB_INTERFACES=:030102:
E: ID_USB_INTERFACE_NUM=01
E: ID_USB_DRIVER=usbhid
E: ID_PATH=pci-0000:00:14.0-usb-0:1:1.1
E: ID_PATH_TAG=pci-0000_00_14_0-usb-0_1_1_1
E: LIBINPUT_DEVICE_GROUP=3/256c/6d:usb-0000:00:14.0-1
E: DEVLINKS=/dev/input/by-id/usb-HUION_Huion_Tablet_KD100-event-if01 /dev/input/by-path/pci-0000:00:14.0-usb-0:1:1.1-event
E: TAGS=:power-switch:
E: CURRENT_TAGS=:power-switch:
```

**Comment by albfan** [comment]
Looks usb dongle exposes a tablet pad, keyboard, key, mouse and touchscreen and then is up to device connected to send events through it.

that's why there's no touchscreen or mouse on wired mode.

**Comment by whot** [comment]
I think a name match here would be sufficient. libwacom is only a lookup library for information on devices detected elsewhere, so it doesn't matter that there are other device nodes with different capabilities.

e.g. libinput will find the `ID_INPUT_TABLET` device and ask libwacom for information on this, but it won't do so for the non-tablet event nodes. IOW as long as we can identify this, and we can by name, we're good.

**Comment by albfan** [comment]
@whot I see libwacom-list-local-devices listing both H950p and HS611 at same time because both match `usb:256c:006d`

Looks problematic as seems there's no way to tell which is the real one, and for GNOME implementation (not business here) only first one is picked. Maybe just offer which to pick will fix it.

But I think when usb wireless is used problem is different, is ok dongle only uses `usb:256c:006d` as it is a generic receiver, but real device is behind that dongle so you need to find which is really the device. As said, if you connect wired, device reports Bus `001 Device 011: ID 256c:006d HUION Huion Tablet_KD100` which looks unique

**Comment by whot** [comment]
note that `libwacom-list-local-devices` is a debugging tool, it simply queries every event node whether there's a match, even if that event node is not a tablet. That's not what GNOME should do (does?), ideally the `ID_INPUT_TABLET` property is compared there or it's pre-filtered by some other part of the stack (e.g. libinput + wayland)

but ultimately whatever solution we have is just kicking the can down the road - until Huion stops re-using USB ids there's very little we can do. IIRC they don't bother updating the device name either so you can buy wildly different devices that are impossible to distinquish from userspace. Maybe changing the kernel drivers to add some identifying information (in `struct input_id`'s version field?) is posible, long term that's the only real solution here.

**Comment by albfan** [comment]
Got what you mean, so 

https://github.com/kurikaesu/userspace-tablet-driver-daemon/blob/210b336b80fd74b7753470b770651969ccd51bff/src/huion_tablet.cpp#L89-L101


needs to be resolved at digimend kernel driver, and exposed in https://docs.huihoo.com/doxygen/linux/kernel/3.7/structinput__id.html

I have use case with several huion devices, so will dig into how GNOME is finding what to map or extend to support several tablets at same time

**Comment by whot** [comment]
Yep, basically that. In the kernel you'd just hardcode each model with a fixed version and hope that it never becomes necessary to actually expose a version field. Userspace will still need to have the same mapping somewhere then, but that's still better than the current guesswork.

**Comment by albfan** [comment]
Adding layout and config for tablet

[huion-kd100.zip](https://github.com/linuxwacom/libwacom/files/7954467/huion-kd100.zip)

Last buttons From M to R still not working, but using https://github.com/DIGImend/usbhid-dump I'm able to see all them so probably something to fix on driver.

Not sure how to setup ring and central button (is supposed that long click enables pop up for mode change, and short click change mode directly, but that needs an overlay to show modes. Is that covered.

I can do a screencast on windows to show this feature if needed.

**Comment by whot** [comment]
Please submit the layout/config file as a proper PR, too hard to review something like this in a zip file. Surprisingly, the zip file spec doesn't accommodate for review comments ;)

In regards to setup: libwacom has zero control over this, libwacom merely says "device has a ring" and to some degree where the button is but everything else is controlled by something else - e.g. the GNOME stack

**Comment by albfan** [comment]
Ouch, yes, I totally forget that, is just I'm trying to connect digimend, libwacom, GNOME and miss the point hehe.

About config and layout, the question was about how to represent that on svg and on config, but that can be review on PR

**Comment by whot** [comment]
I think this can be closed with #470 merged now. The remaining issues appear to be outside libwacom's bits
